hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/opt/logical/FunctionLogicalRules.java

```diff
@@ -37,12 +39,12 @@
 
 import static java.util.stream.Collectors.toList;
 
-final class FullFunctionScanLogicalRules {
+final class FunctionLogicalRules {
```

Comment:
Window aggregation queries produce plans containing this: 
```
AggregateLogicalRel(group=[{0}], EXPR$1=[MIN($1)], EXPR$2=[MIN($2)])
  ProjectLogicalRel(window_start=[$3], name=[$1], distance=[$2])
    SlidingWindowLogicalRel(invocation=[TUMBLE($2, DESCRIPTOR($0), 2:INTERVAL DAY TO SECOND)], rowType=[RecordType(TIMESTAMP WITH TIME ZONE ts, VARCHAR name, INTEGER distance, TIMESTAMP WITH TIME ZONE window_start, TIMESTAMP WITH TIME ZONE window_end)])
```
Each of these steps is backed by a separate vertex (Aggregate by 2 of them actually), however, all these three steps can be executed by `SlidingWindowP`. Maybe there can be 2 rules: one merging the `Project` into `SlidingWindowRel`, and the other merging `Aggregate` and `SlidingWindowRel` into `SlidingWindowAggrRel`.

It can maybe be done in a separate PR.

Reply:
#19882